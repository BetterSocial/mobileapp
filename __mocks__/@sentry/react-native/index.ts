export const addBreadcrumb = jest.fn();
