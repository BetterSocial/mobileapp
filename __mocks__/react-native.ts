import * as RN from 'react-native';

RN.Animated.timing = () => ({
    start: () => jest.fn()
})

module.exports = RN