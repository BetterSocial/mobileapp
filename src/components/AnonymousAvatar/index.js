import * as React from 'react';
import {Image, StyleSheet, Text, View} from 'react-native';

import AnonymousProfile from '../../assets/images/AnonymousProfile.png';
import dimen from '../../utils/dimen';
import {POST_VERSION} from '../../utils/constants';

export const styles = StyleSheet.create({
  imageAnonimity: {
    marginRight: 0,
    width: dimen.size.FEED_HEADER_IMAGE_RADIUS,
    height: dimen.size.FEED_HEADER_IMAGE_RADIUS
  },
  avatarV2Background: (backgroundColor, radius) => ({
    width: radius,
    height: radius,
    borderRadius: radius / 2,
    backgroundColor,
    justifyContent: 'center'
  }),
  avatarV2Emoji: (emojiRadius) => ({
    fontSize: emojiRadius,
    textAlign: 'center',
    alignSelf: 'center'
  })
});

/**
 * @typedef {Object} AnonymousAvatarComponentProps
 * @property {number} [radius=dimen.size.FEED_HEADER_IMAGE_RADIUS]
 * @property {number} [emojiRadius=dimen.size.FEED_HEADER_ANONYMOUS_IMAGE_RADIUS]
 * @property {string} version
 * @property {import('react-native').StyleProp} containerStyle
 * @property {AnonUserInfoTypes} anonUserInfo
 */

/**
 *
 * @param {AnonymousAvatarComponentProps} props
 */
const AnonymousAvatar = (props) => {
  const {
    version,
    anonUserInfo,
    containerStyle = {},
    radius = dimen.size.FEED_HEADER_IMAGE_RADIUS,
    emojiRadius = dimen.size.FEED_HEADER_ANONYMOUS_IMAGE_RADIUS
  } = props;

  if (version >= POST_VERSION) {
    return (
      <View
        testID="newVersion"
        style={{...styles.avatarV2Background(anonUserInfo.colorCode, radius), ...containerStyle}}>
        <Text style={styles.avatarV2Emoji(emojiRadius)}>{anonUserInfo.emojiCode}</Text>
      </View>
    );
  }

  return (
    <Image
      source={AnonymousProfile}
      width={radius}
      height={radius}
      style={{...styles.imageAnonimity, ...containerStyle}}
    />
  );
};

export default AnonymousAvatar;
