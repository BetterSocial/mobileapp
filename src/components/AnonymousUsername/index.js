import * as React from 'react';
import {StyleSheet, Text} from 'react-native';

import {POST_VERSION} from '../../utils/constants';
import {colors} from '../../utils/colors';
import {fonts} from '../../utils/fonts';

const styles = StyleSheet.create({
  feedUsername: {
    fontFamily: fonts.inter[600],
    fontWeight: 'bold',
    fontSize: 14,
    lineHeight: 16.94,
    color: colors.black,
    flex: 1
  }
});

/**
 * @typedef {Object} AnonymousUsernameComponentProps
 * @property {string} version
 * @property {AnonUserInfoTypes} anonUserInfo
 */

/**
 *
 * @param {AnonymousUsernameComponentProps} props
 */
const AnonymousUsername = (props) => {
  const {version, anonUserInfo} = props;

  if (version >= POST_VERSION) {
    return (
      <Text
        testID="newVersion"
        style={styles.feedUsername}>{`${anonUserInfo.colorName} ${anonUserInfo.emojiName}`}</Text>
    );
  }

  return <Text style={styles.feedUsername}>Anonymous</Text>;
};

export default AnonymousUsername;
