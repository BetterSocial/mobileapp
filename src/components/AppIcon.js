import React from 'react';
import {Image, StyleSheet} from 'react-native';

function AppIcon() {
  return (
    <Image
      source={require('./../assets/ping.png')}
      style={styles.logo}
      resizeMode="cover"
    />
  );
}

const styles = StyleSheet.create({
  logo: {
    width: 41,
    height: 52,
  },
});

export default AppIcon;
