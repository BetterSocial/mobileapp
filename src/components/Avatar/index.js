import * as React from 'react';
import {View, StyleSheet, Image} from 'react-native';

const index = ({image, style}) => {
  return (
    <View style={[styles.wrapperImage, style]}>
      <Image
        source={{
          uri: image
            ? image
            : 'https://res.cloudinary.com/hpjivutj2/image/upload/v1617245336/Frame_66_1_xgvszh.png',
        }}
        style={[styles.image, StyleSheet.absoluteFillObject]}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  wrapperImage: {
    borderRadius: 45,
    borderWidth: 0.2,
    borderColor: 'rgba(0,0,0,0.5)',
    width: 46,
    height: 46,
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {
    height: '100%',
    width: '100%',
    borderRadius: 45,
  },
});

export default index;
