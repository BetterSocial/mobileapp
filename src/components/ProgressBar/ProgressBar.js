import * as React from 'react';
import {LogBox, View, StyleSheet, Animated} from 'react-native';
import {COLORS} from '../../utils/theme';

function useInterval(callback, delay) {
  const savedCallback = React.useRef();

  React.useEffect(() => {
    savedCallback.current = callback;
  }, [callback]);

  React.useEffect(() => {
    LogBox.ignoreLogs(['Animated: `useNativeDriver`']);
  }, []);

  // eslint-disable-next-line consistent-return
  React.useEffect(() => {
    function tick() {
      savedCallback.current();
    }
    if (delay !== null) {
      const id = setInterval(tick, delay);
      return () => clearInterval(id);
    }
  }, [delay]);
}

const ProgressBar = (props) => {
  const animation = React.useRef(new Animated.Value(0));
  const [progress, setProgress] = React.useState(50);
  useInterval(() => {
    if (progress < 100 && !props.isStatic) {
      setProgress(progress + 5);
    }
  }, 1000);

  React.useEffect(() => {
    Animated.timing(animation.current, {
      toValue: progress,
      duration: 100
    }).start();
  }, [progress]);

  React.useEffect(() => {
    setProgress(props.value);
  }, [props.value]);

  const width = animation.current.interpolate({
    inputRange: [0, 100],
    outputRange: ['0%', '100%'],
    extrapolate: 'clamp'
  });

  return (
    <View style={styles.container}>
      <View style={styles.progressBar}>
        <Animated.View style={([StyleSheet.absoluteFill], styles.animated(width))} />
      </View>
    </View>
  );
};

export default ProgressBar;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: COLORS.porcelain,
    borderRadius: 5
  },
  progressBar: {
    flexDirection: 'row',
    height: 8,
    width: '100%',
    backgroundColor: COLORS.gray100,
    borderRadius: 5
  },
  animated: (width) => ({
    backgroundColor: COLORS.signed_primary,
    borderRadius: 5,
    width
  })
});
