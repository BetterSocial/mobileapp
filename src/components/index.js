import Avatar from './Avatar';
import SingleSidedShadowBox from './SingleSidedShadowBox';
import Footer from './Footer/Footer';

export {Avatar, SingleSidedShadowBox, Footer};
