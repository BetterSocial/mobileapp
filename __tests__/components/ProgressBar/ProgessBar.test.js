import React from 'react';
import {render} from '@testing-library/react-native';
import {Animated} from 'react-native';
import {ProgressBar} from '../../../src/components/ProgressBar';

describe('Progress Bar should run correctly', () => {
  const mockInterpolate = jest.fn();
  const mockTiming = jest.fn();
  beforeEach(() => {
    jest
      .spyOn(React, 'useRef')
      .mockImplementation(() => ({current: {interpolate: mockInterpolate}}));
    jest.spyOn(Animated, 'timing').mockImplementation(() => ({start: mockTiming}));
  });
  it('should match with snapshot', () => {
    const {toJSON} = render(<ProgressBar />);
    expect(toJSON).toMatchSnapshot();
  });
});
