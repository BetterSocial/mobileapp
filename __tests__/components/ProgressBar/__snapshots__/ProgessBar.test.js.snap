// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Bar should run correctly should match with snapshot 1`] = `[Function]`;
